base/common/DateLUTImpl.h

```diff
@@ -673,6 +673,10 @@ class DateLUTImpl
         if (unlikely(year < DATE_LUT_MIN_YEAR || year > DATE_LUT_MAX_YEAR || month < 1 || month > 12 || day_of_month < 1 || day_of_month > 31))
             return DayNum(0); // TODO (nemkov, DateTime64 phase 2): implement creating real date for year outside of LUT range.
 
+        // The day after 2106-02-07 will not stored fully as struct Values, so just overflow it as 0
+        if (unlikely(year == DATE_LUT_MAX_YEAR && (month > 2 || day_of_month > 7)))
```

Comment:
But this condition is not correct. It will overflow also all dates in `[2106-01-08; 2106-01-31].

Reply:
O, yes, it's wrong, I will fix this